protocols/collective_aggregation_protocol.go

```diff
@@ -195,11 +196,15 @@ func (p *CollectiveAggregationProtocol) Dispatch() error {
 
 // Announce forwarding down the tree.
 func (p *CollectiveAggregationProtocol) aggregationAnnouncementPhase() error {
-	dataReferenceMessage := <-p.DataReferenceChannel
-	if !p.IsLeaf() {
-		if err := p.SendToChildren(&dataReferenceMessage.DataReferenceMessage); err != nil {
-			return errors.New("Error sending <DataReferenceMessage>:" + err.Error())
+	select {
+	case dataReferenceMessage := <-p.DataReferenceChannel:
+		if !p.IsLeaf() {
```

Comment:
Unneded if, `SendToChildren` when having none should work.

Reply:
31c25d5

Reply:
Arf, I misexplained it; the `if !p.IsLeaf()` is unnecessary, as a Leaf sending to it's children is a NOP.
And you're droping the `error` here.

Reply:
ahh I did not get it :P 
Ok it's fixed: f3812c3